dissect/target/plugin.py

```diff
@@ -527,7 +528,7 @@ def decorator(obj):
     return decorator
 
 
-def plugins(osfilter: str = None) -> Iterator[PluginDescriptor]:
+def plugins(osfilter: type[OSPlugin] = None) -> Iterator[PluginDescriptor]:
```

Comment:
```suggestion
def plugins(osfilter: Optional[type[OSPlugin]] = None) -> Iterator[PluginDescriptor]:
```

Does this not also break other APIs? This function is used in a number of places if I recall correctly.